lightning/src/routing/router.rs

```diff
@@ -99,7 +99,6 @@ impl<G: Deref<Target = NetworkGraph<L>> + Clone, L: Deref, S: Deref, SP: Sized,
 
 		let network_graph = self.network_graph.deref().read_only();
 		let paths = first_hops.into_iter()
-			.filter(|details| details.is_public)
```

Comment:
Needed to drop this since we want two-hop paths that contain an unannounced channel between the intro node and us.